tools/dev/loki-boltdb-storage-s3/docker-compose.yml

```diff
@@ -0,0 +1,219 @@
+version: '3.4'
+services:
+  consul:
+    logging: &logging
+      driver: loki-compose
+      options:
+        loki-url: "http://localhost:8001/loki/api/v1/push"
+    image: consul
+    command: [ "agent", "-dev" ,"-client=0.0.0.0", "-log-level=info" ]
+    ports:
+      - 8500:8500
+
+  minio:
+    logging:
+      <<: *logging
+    image: minio/minio
+    entrypoint: sh
+    command: -c 'mkdir -p /data/loki && /usr/bin/minio server /data'
+    environment:
+      - MINIO_ACCESS_KEY=loki
+      - MINIO_SECRET_KEY=supersecret
+    ports:
+      - 9000:9000
+    volumes:
+      - .data-minio:/data:delegated
+
+  memcached:
+    logging:
+      <<: *logging
+    image: memcached:1.6
+
+  jaeger:
+    logging:
+      <<: *logging
+    image: jaegertracing/all-in-one
+    ports:
+      - 16686:16686
+      - "14268"
+
+  distributor:
+    logging:
+      <<: *logging
+    build:
+      context:    .
+      dockerfile: dev.dockerfile
+    image: loki
+    command: ["sh", "-c", "sleep 3 && exec ./dlv exec ./loki --listen=:18001 --headless=true --api-version=2 --accept-multiclient --continue -- -config.file=./config/loki.yaml -target=distributor -server.http-listen-port=8001 -server.grpc-listen-port=9001"]
```

Comment:
What is the idea behind `sleep 3`?

Reply:
Just to leave enough time for minio and consul to be up and running.

Reply:
Isn't the image restarted if it crashes? This was the annoying bit about Docker compose. It was not easy to time dependencies.

Reply:
I don't think Loki crashes for this and not it doesn't restart but it would only refuse to start if the config is wrong, not if it can't connect.